src/clientsettingsdlg.cpp

```diff
@@ -1009,17 +1016,37 @@ void CClientSettingsDlg::OnEnableOPUS64StateChanged ( int value )
 
 void CClientSettingsDlg::OnFeedbackDetectionChanged ( int value ) { pSettings->bEnableFeedbackDetection = value == Qt::Checked; }
 
-void CClientSettingsDlg::OnCustomDirectoriesEditingFinished()
+void CClientSettingsDlg::OnDeleteCustomDirectoriesClicked()
```

Comment:
(In fact, both signal handlers could call a common method passing an argument to indicate "delete" or "add", so that the only change would be working out whether an entry was being deleted by seeing if the field is empty.  Adding would have to prevent adding empty values, of course.)